egs/wsj/s5/steps/nnet3/multi/make_configs.py

```diff
@@ -0,0 +1,678 @@
+                        help="The layer number to add bottleneck layer,"
+                        "if < 0, means this layer is not needed in network.",
+                        default=-1)
+    parser.add_argument("--bottleneck-dim", type=int,
+                        help="The bottleneck layer dimension in TDNN network e.g. 42.",
+                        default=40)
+    print(' '.join(sys.argv))
+
+    args = parser.parse_args()
+    args = CheckArgs(args)
+
+    return args
+
+def CheckArgs(args):
+    if not os.path.exists(args.config_dir):
+        os.makedirs(args.config_dir)
+
+    ## Check arguments.
+    if args.feat_dir is not None:
+        args.feat_dim = nnet3_train_lib.GetFeatDim(args.feat_dir)
+
+    if args.ali_dir is not None:
+        args.num_targets = nnet3_train_lib.GetNumberOfLeaves(args.ali_dir)
+    elif args.tree_dir is not None:
+        args.num_targets = chain_lib.GetNumberOfLeaves(args.tree_dir)
+
+    if args.ivector_dir is not None:
+        args.ivector_dim = nnet3_train_lib.GetIvectorDim(args.ivector_dir)
+
+    if not args.feat_dim > 0:
+        raise Exception("feat-dim has to be postive")
+
+    if not args.num_targets > 0:
+        if args.num_multiple_targets is None:
+          print(args.num_targets)
+          raise Exception("num_targets or num_multiple_targets has to be positive")
+    
+    if not args.ivector_dim >= 0:
+        raise Exception("ivector-dim has to be non-negative")
+
+    if (args.subset_dim < 0):
+        raise Exception("--subset-dim has to be non-negative")
+    if (args.pool_window is not None) and (args.pool_window <= 0):
+        raise Exception("--pool-window has to be positive")
+
+    if not args.relu_dim is None:
+        if not args.pnorm_input_dim is None or not args.pnorm_output_dim is None:
+            raise Exception("--relu-dim argument not compatible with "
+                            "--pnorm-input-dim or --pnorm-output-dim options");
+        args.nonlin_input_dim = args.relu_dim
+        args.nonlin_output_dim = args.relu_dim
+    else:
+        if not args.pnorm_input_dim > 0 or not args.pnorm_output_dim > 0:
+            raise Exception("--relu-dim not set, so expected --pnorm-input-dim and "
+                            "--pnorm-output-dim to be provided.");
+        args.nonlin_input_dim = args.pnorm_input_dim
+        args.nonlin_output_dim = args.pnorm_output_dim
+
+    if args.add_final_sigmoid and args.include_log_softmax:
+        raise Exception("--include-log-softmax and --add-final-sigmoid cannot both be true.")
+
+    if args.xent_separate_forward_affine and args.add_final_sigmoid:
+        raise Exception("It does not make sense to have --add-final-sigmoid=true when xent-separate-forward-affine is true")
+
+    if args.add_lda and args.cnn_layer is not None:
+        args.add_lda = False
+        warnings.warn("--add-lda is set to false as CNN layers are used.")
+
+    return args
+
+def AddPerDimAffineLayer(config_lines, name, input, input_window):
```

Comment:
This config generator script seems to have been copied from an older tdnn config generator. The new script is eliminates a lot of these options.